pandas/core/series.py

```diff
@@ -2864,35 +2867,46 @@ def map(self, arg, na_action=None):
         1    foo
         2    bar
         3    baz
+        dtype: object
```

Comment:
I think these examples would be clearer if instead of `foo`, `bar`, `one`, `two`... we could find a more real-world example.

Also, I think it should be useful to show the behaviour of `map` with a `dict` or `Series` that doesn't have all the values in the `Series`.